external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWriter.scala

```diff
@@ -101,4 +86,16 @@ private[kafka010] object KafkaWriter extends Logging {
         finallyBlock = writeTask.close())
     }
   }
+
+  private def validateAttribute(
+      schema: Seq[Attribute],
+      attrName: String,
+      desired: Seq[DataType])(
+      defaultFn: () => Expression): Unit = {
+    val attr = schema.find(_.name == attrName).getOrElse(defaultFn())
+    if (!desired.exists(_.sameType(attr.dataType))) {
+      throw new AnalysisException(s"$attrName attribute type must be a " +
```

Comment:
I think it would be helpful to print the actual type and the expected types, just like in the previous case.